src/pages/debts/index.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useRef, useState } from "react";
+import { useRecoilState } from "recoil";
+import {
+  DebtFormI,
+  DebtI,
+  selectedDebtState,
+  useDebts,
+} from "../../data/debts";
+import { useCategories } from "../../data/categories";
+import { userState } from "../../data/authentication";
+import { Transition } from "@headlessui/react";
+import classNames from "classnames";
+import { AiFillEdit } from "react-icons/ai";
+import { Card } from "../../components/cards/card";
+import { DeleteCardModal } from "../../components/cards/delete-card";
+import { DebtModal } from "./debts-modal";
+
+export const Debts: React.FC = () => {
+  const [selectedDebt, setSelectedDebt] = useRecoilState(selectedDebtState);
+  const [user] = useRecoilState(userState);
+  const { getDebts, debtsList, deleteDebt } = useDebts();
+  const { getCategories } = useCategories();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const timer = useRef<ReturnType<typeof setTimeout>>();
+  const selectedContainer = useRef<HTMLDivElement | null>(null);
+
+  function openModal() {
+    setIsModalOpen(true);
+  }
+
+  function closeModal() {
+    setIsModalOpen(false);
+    setSelectedDebt(null);
+  }
+
+  useEffect(() => {
+    getCategories();
+    getDebts();
+  }, [user]);
```

Comment:
This will be executed twice with the first render when user is null and then when user has a value